packages/vite/src/executors/test/vitest.impl.ts

```diff
@@ -0,0 +1,77 @@
+import { File, Reporter } from 'vitest';
+import { VitestExecutorSchema } from './schema';
+
+class NxReporter implements Reporter {
+  deferred: {
+    promise: Promise<boolean>;
+    resolve: (val: boolean) => void;
+  };
+
+  constructor(private watch: boolean) {
+    this.setupDeferred();
+  }
+
+  async *[Symbol.asyncIterator]() {
+    do {
+      const hasErrors = await this.deferred.promise;
+      yield { hasErrors };
+      this.setupDeferred();
+    } while (this.watch);
+  }
+
+  private setupDeferred() {
+    let resolve: (val: boolean) => void;
+    this.deferred = {
+      promise: new Promise((res) => {
+        resolve = res;
+      }),
+      resolve,
+    };
+  }
+
+  onFinished(files: File[], errors?: unknown[]) {
+    const hasErrors =
+      files.some((f) => f.result?.state === 'fail') || errors?.length > 0;
+    this.deferred.resolve(hasErrors);
+  }
+}
+
+export default async function* runExecutor(options: VitestExecutorSchema) {
+  const { startVitest } = await (Function(
+    'return import("vitest/node")'
+  )() as Promise<typeof import('vitest/node')>);
+
+  const nxReporter = new NxReporter(options.watch);
```

Comment:
Is this custom reporter needed? I don't think `watch=true` would need to exit with non-zero status, so if `startVitest` or another function can return the status to us then that should be enough.

Reply:
For context when others come across this:
When running watch, there's certain situations where the process will always exit with `1`, so we need to force the process to exit with `0` if there's no errors within the watch process.